jib-core/src/main/java/com/google/cloud/tools/jib/image/Image.java

```diff
@@ -49,18 +50,14 @@
     }
 
     /**
-     * Sets the environment with a map from environment variable names to values.
+     * Adds a map of environment variables to the current map.
      *
      * @param environment the map of environment variables
      * @return this
      */
-    public Builder<T> setEnvironment(@Nullable Map<String, String> environment) {
-      if (environment == null) {
-        this.environmentBuilder = ImmutableList.builder();
-      } else {
-        for (Map.Entry<String, String> environmentVariable : environment.entrySet()) {
-          setEnvironmentVariable(environmentVariable.getKey(), environmentVariable.getValue());
-        }
+    public Builder<T> addEnvironment(@Nullable Map<String, String> environment) {
+      if (environment != null) {
+        this.environmentBuilder.putAll(ImmutableMap.copyOf(environment));
```

Comment:
I'm thinking this is no different from and just could be `environmentBuilder.putAll(environment)`, am I right? You are just putting references to objects (`String`s in this case) into the builder, not like you are doing a deep copy of `String` instances. (And `String`s are immutable, so there is no point of copying, actually.) Am I right?

Reply:
Sounds right to me.